rows.go

```diff
@@ -46,6 +46,14 @@ func (r *rows) Columns() []string {
 	return columns
 }
 
+func (r *rows) ColumnTypeDatabaseTypeName(index int) (typeName string) {
```

Comment:
I prefer not using named returns unless much simpler. It's nice to be explicit that an empty string is returned if the type is unknown (for whatever reason...). Mind changing that before merging?